From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Thomas Staudinger <Staudi.Kaos@gmail.com>
Date: Sat, 11 May 2024 22:32:02 +0200
Subject: [PATCH] gtksourceview: Add support for new libgedit-gtksourceview xml
 scheme

The `kind` attribute is now mandatory, `version` and `underline-color` removed

Signed-off-by: Thomas Staudinger <Staudi.Kaos@gmail.com>
---
 gtksourceview/gtksourceview/dark.xml.in    | 4 ++--
 gtksourceview/gtksourceview/default.xml.in | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/gtksourceview/gtksourceview/dark.xml.in b/gtksourceview/gtksourceview/dark.xml.in
index 8b7584839..ff3b67d4f 100644
--- a/gtksourceview/gtksourceview/dark.xml.in
+++ b/gtksourceview/gtksourceview/dark.xml.in
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<style-scheme id="@FlavourThemeName@" _name="@FlavourThemeName@" version="1.0">
+<style-scheme id="@FlavourThemeName@" _name="@FlavourThemeName@" kind="dark">
   <_description>A GtkSourceView style to match @FlavourThemeName@</_description>
 
   <!-- Yaru palette -->
@@ -94,7 +94,7 @@
   <style name="def:strong-emphasis"        bold="true"/>
   <style name="def:type"                   foreground="blue_2" bold="true"/>
   <style name="def:underlined"             underline="single"/>
-  <style name="def:warning"                underline="error" underline-color="yellow_4"/>
+  <style name="def:warning"                underline="error"/>
   
   <!-- C# -->
   <style name="c-sharp:format"             foreground="purple_2"/>
diff --git a/gtksourceview/gtksourceview/default.xml.in b/gtksourceview/gtksourceview/default.xml.in
index e6cad5c5d..7d753e037 100644
--- a/gtksourceview/gtksourceview/default.xml.in
+++ b/gtksourceview/gtksourceview/default.xml.in
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<style-scheme id="@FlavourThemeName@" _name="@FlavourThemeName@" version="1.0">
+<style-scheme id="@FlavourThemeName@" _name="@FlavourThemeName@" kind="light">
   <_description>A GtkSourceView style to match @FlavourThemeName@</_description>
 
   <!-- Yaru palette -->
@@ -92,7 +92,7 @@
   <style name="def:strong-emphasis"        bold="true"/>
   <style name="def:type"                   foreground="blue_4" bold="true"/>
   <style name="def:underlined"             underline="single"/>
-  <style name="def:warning"                underline="error" underline-color="yellow_4"/>
+  <style name="def:warning"                underline="error"/>
 
   <!-- C# -->
   <style name="c-sharp:format"             foreground="purple_4"/>
